docs/ADR-74-add-emote-schema.md

```diff
@@ -0,0 +1,142 @@
+
+// hypothetical future ADR 102
+export type EmoteADR102 = BaseItem & { emoteDataADR102: EmoteDataADR102 };
+export type Emote = EmoteADR74 | EmoteADR102;
+```
+
+##### 3. Create new EmoteCategory and EmoteRepresentation types specific for Emotes.
+
+Although they're pretty similar, they are different to the ones for Wearables and may diverge more in the future. They will also be included in the versioned properties.
+One caveat, _EmoteCategory_ is imported from _src/dapps_ but it's not versioned as there only matters the latest version in that context. For that, we propose to change the dependency to the other way, the latest version is imported in _src/dapps from src/platform/_.
+If a new category is added, it will be added to _EmoteCategory_ and a new type called _EmoteCategoryADR74_ will be created to validate emotes before this new change.
+
+```typescript
+// <src/platform/…>
+export type EmoteRepresentationADR74 = {
+ bodyShapes: BodyShape[]
+ mainFile: string
+ contents: string[]
+}
+
+export enum EmoteCategory {
+ SIMPLE = 'simple',
+ LOOP = 'loop'
+}
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+// <src/dapps/…>
+import { EmoteCategory } from '.../src/platform/...'
+```
+
+##### 4. Stop exporting qualitative types.
+
+(Breaking change) Stop exporting the ThirdPartyWearable, StandardWearable and don't export the analogues for Emote. Instead, the exported types will be Wearable, Emote, StandardProps, ThirdPartyProps and the functions isStandard, isThirdParty. This way, we will prevent the explosion of the combination of those types (and future properties added):
+
+```typescript
+export function isStandard(item: Item): item is Item & StandardProps {. . .}
+export function isThirdParty(item: Item): item is Item & ThirdPartyProps {. . .}
+// Instead of
+export type ThirdPartyEmote
+export type StandardEmote
+export type ThirdPartyWearable
+export type StandardWearable
+// Now imaging that we add a new set of properties, let's call it t'Magical' that is combinable with the others:
+export type ThirdPartyEmote
+export type StandardEmote
+export type ThirdPartyWearable
+export type StandardWearable
+export type MagicalThirdPartyEmote
+export type MagicalStandardEmote
+export type MagicalThirdPartyWearable
+export type MagicalStandardWearable
+// and so on ...
+```
+
+## Benefit
+
+The bigger benefit is that the validation of schemas over the time becomes somehow trivial\*:
+
+```typescript
+function validate(emote) {
+   adrXX = identifyADR(emote.timestamp)
+   if (emote does not have the field 'emoteData' + ardXX) return false
+   Emote.schema.validate(emote)
+}
+```
+
+\*under the assumption that the common properties do not change.
+This would give us the possibility to iterate schema changes quicker and also easily identify the versions over the time along with its reasons explained in the specific ADR.
```

Comment:
```suggestion
This would give the possibility to iterate schema changes quicker and also easily identify the versions over the time along with its reasons explained in the specific ADR.
```